File: src/org/ggp/base/util/Pair.java
package org.ggp.base.util;

import java.util.Map;

class Pair <L, R> {
    def final L left;
    def final R right;

    private Pair(L left, R right):
        this.left = left;
        this.right = right;

    def static <L, R> Pair<L, R> of(L left, R right):
        return new Pair<L, R>(left, right);

    def static <L, R> Pair<L, R> from(Map.Entry<L, R> entry):
        return of(entry.getKey(), entry.getValue());

    def hashCode():  # int
        final int prime = 31;
        int result = 1;
        result = prime * result + ((left == null) ? 0 : left.hashCode());
        result = prime * result + ((right == null) ? 0 : right.hashCode());
        return result;

    def bool equals(Object obj):
        if (this == obj)
            return true;
        if (obj == null)
            return false;
        if (getClass() != obj.getClass())
            return false;
        Pair<?, ?> other = (Pair<?, ?>) obj;
        if (left == null):
            if (other.left != null)
                return false;
		} else if (!left.equals(other.left))
            return false;
        if (right == null):
            if (other.right != null)
                return false;
		} else if (!right.equals(other.right))
            return false;
        return true;

    def toString():  # String
        return "<" + left + ", " + right + ">";

File: src/org/ggp/base/util/Pair.py
#!/usr/bin/env python
""" generated source for module Pair """
# package: org.ggp.base.util
import java.util.Map

class Pair(object):
    """ generated source for class Pair """
    left = L()
    right = R()

    def __init__(self, left, right):
        """ generated source for method __init__ """
        self.left = left
        self.right = right

    @classmethod
    def of(cls, left, right):
        """ generated source for method of """
        return Pair(left, right)

    @classmethod
    def from_(cls, entry):
        """ generated source for method from_ """
        return cls.of(entry.getKey(), entry.getValue())

    def hashCode(self):
        """ generated source for method hashCode """
        prime = 31
        result = 1
        result = prime * result + (0 if (self.left == None) else self.left.hashCode())
        result = prime * result + (0 if (self.right == None) else self.right.hashCode())
        return result

    def equals(self, obj):
        """ generated source for method equals """
        if self == obj:
            return True
        if obj == None:
            return False
        if getClass() != obj.__class__:
            return False
        other = obj
        if self.left == None:
            if other.left != None:
                return False
        elif not self.left == other.left:
            return False
        if self.right == None:
            if other.right != None:
                return False
        elif not self.right == other.right:
            return False
        return True

    def __str__(self):
        """ generated source for method toString """
        return "<" + self.left + ", " + self.right + ">"

